WorkflowTesting/Sources/RenderTesterResult.swift

```diff
@@ -41,6 +43,22 @@ public struct RenderTesterResult<WorkflowType: Workflow> {
         return self
     }
 
+    /// Exhaustive state testing against the initial state.
+    /// - Parameters:
+    ///   - changes: A function that receives the initial state
+    ///   and is expected to mutate it to match the new state.
+    @discardableResult
+    public func assertState(
+        file: StaticString = #file,
+        line: UInt = #line,
+        changes: (inout WorkflowType.State) throws -> Void
+    ) rethrows -> RenderTesterResult<WorkflowType> where WorkflowType.State: Equatable {
+        var initialState = self.initialState
+        try changes(&initialState)
+        XCTAssertEqual(initialState, state, "Initial state did not match new state", file: file, line: line)
```

Comment:
this error message seems a little confusing to me. we don't actually expect the 'initial state' to match the new state right? we expect the explicitly transformed state to match. not sure what the best wording to communicate this is though...

Reply:
hmm yeah this doesn't make sense... perhaps something more vague like 

> Expected state does not match

Reply:
also perhaps I should reverse the order here

```
XCTAssertEqual(state, initialState
```